sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -325,19 +286,36 @@ def open_db_connections(self, db_key, db_name=None, is_default=True, key_prefix=
         except Exception as e:
             error_message = self.test_network_connectivity()
             tcp_connection_status = error_message if error_message else "OK"
-            message = "Unable to connect to SQL Server (host={} database={}). TCP-connection({}). Exception: {}".format(
-                host, database, tcp_connection_status, _format_connection_exception(e)
-            )
+            exception_msg, conn_warn_msg = format_connection_exception(e, driver)
+            if tcp_connection_status != "OK" and conn_warn_msg is ConnectionErrWarning.unknown:
+                conn_warn_msg = ConnectionErrWarning.tcp_connection_failed
 
             password = self.instance.get('password')
             if password is not None:
-                message = message.replace(password, "*" * 6)
-
-            self.service_check_handler(AgentCheck.CRITICAL, host, database, message, is_default=is_default)
+                exception_msg = exception_msg.replace(password, "*" * 6)
+
+            check_err_message = error_with_tags(
+                "Unable to connect to SQL Server, see %s#%s for more details on how to debug this issue. "
+                "TCP-connection(%s), Exception: %s",
+                SUPPORT_LINK,
+                conn_warn_msg.value,
+                tcp_connection_status,
+                exception_msg,
+                host=self._check.resolved_hostname,
+                database=database,
+                code=conn_warn_msg.value,
+                connector=self.connector,
+                driver=driver,
+            )
+            self.service_check_handler(
```

Comment:
We talked about this "offline" but if we're going to leave this call, I think we should put a comment that the critical status setting is basically unused because what's effectively setting the status is the exception that gets raised and bubbles up to the base check.